python/lsst/daf/butler/formatters/parquetFormatter.py

```diff
@@ -0,0 +1,206 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("ParquetFormatter", )
+
+import json
+import re
+import collections.abc
+import itertools
+import copy
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
+
+from lsst.daf.butler.core.utils import iterable
+from lsst.daf.butler import Formatter, Location
+
+try:
```

Comment:
Formatters are meant to only be imported when you need them so I don't think we need the try block here do we?

Reply:
Generally, yes, that sounds reasonable, but do I need to guard against `pytest-flake8` until the dust settles there?

Reply:
Oh. I had forgotten about that annoyance. But then how does daf_butler work at all? python/lsst/daf/butler/assemblers/exposureAssembler.py imports afw for example.

Reply:
Maybe it's okay when there's no `__init__.py`?  I was just being paranoid because I don't understand how it finds things.

Reply:
If I remove the `try` block and it passes Travis/Jenkins, is that sufficient to know it's safe?  If not I might just keep the paranoia.

Reply:
Fundamentally we don't understand why sometimes pytest-flake8 is importing code that it didn't import before (especially given that we didn't change flake8 or pytest-flake8 when we switched to conda pytest).

I'd really rather we didn't go through and put lots of try blocks in every formatter. It makes them untidy and is not required by the butler infrastructure and will likely lead to everyone adding try blocks defensively for no reason. 

I just ran butler tests without afw setup and everything passes (so that's good).

Reply:
I removed the try block and Travis and local scons are still happy.  Launching Jenkins now.